test/unit/plugins/rest_spec.js

```diff
@@ -646,4 +647,233 @@ describe("The Rest plugin", function () {
+
+			var responseCode;
+			var updatedInstance;
+			var instanceID;
+			before(function () {
+
+				return instances.createInstance()
+				.then(function (instance) {
+					instanceID = instance.id;
+					return instances.getInstance({ id : instance.id });
+				})
+				.then(function (instance) {
+					updatedInstance = new Instance({
+						id       : instance.id,
+						ami      : instance.ami,
+						type     : instance.type,
+						state    : "failed",
+						uri      : instance.uri,
+						revision : 1
+					});
+					var request = new Request("PUT", "/v1/instances/" + instance.id).mime("application/json")
+					.payload({
+						ami      : instance.ami,
+						type     : instance.type,
+						uri      : instance.uri,
+						state    : "failed",
+						revision : 0
+					});
+					return request.inject(server);
+				})
+				.then(function (response) {
+					responseCode = response.statusCode;
+				});
+			});
+
+			it("the status is set to failed", function () {
+				return instances.getInstance({ id : instanceID })
+				.then(function (response) {
+					expect(response).to.deep.equal(updatedInstance);
+					expect(responseCode).to.equal(200);
+				});
+			});
+		});
+
+		describe("when the instance ID doesn't exist", function () {
+
+			var result;
+
+			before(function () {
+				var request = new Request("PUT", "/v1/instances/" + INVALID_QUERY).mime("application/json")
+				.payload({
+					ami      : "ami-d05e75b8",
+					type     : "t2.micro",
+					uri      : null,
+					state    : "failed",
+					revision : 0
+				});
+				return request.inject(server)
+				.then(function (response) {
+					result = response;
+				});
+			});
+
+			it("a 404 error is thrown", function () {
+				expect(result.statusCode).to.equal(404);
+			});
+		});
+
+		describe("when the payload is malformed", function () {
+
+			var result;
+
+			before(function () {
+				var request = new Request("PUT", "/v1/instances/" + INVALID_QUERY).mime("application/json")
+				.payload({
+					ami      : [ "ami-d05e75b8" ],
+					type     : "t2.micro",
+					uri      : null,
+					state    : "failed",
+					revision : 0
+				});
+				return request.inject(server)
+				.then(function (response) {
+					result = response;
+				});
+			});
+
+			it("a 400 error is thrown", function () {
+				expect(result.statusCode).to.equal(400);
+			});
+		});
+
+		describe("when two requests are made simultaneously", function () {
+
+			var firstResponse;
+			var secondResponse;
+			var instanceID;
+
+			before(function () {
+				return instances.createInstance()
+				.then(function (instance) {
+					instanceID = instance.id;
+					return instances.getInstance({ id : instance.id });
+				})
+				.then(function (instance) {
+					var firstRequest = new Request("PUT", "/v1/instances/" + instance.id).mime("application/json")
+					.payload({
+						ami      : instance.ami,
+						type     : instance.type,
+						uri      : null,
+						state    : "ready",
+						revision : 0
```

Comment:
Similar to above, would it make more sense to pull this off of the `instance` object?